### DISCUSSION TITLE
[BUG]:当配置被新增和删除时,未被改动的配置有非预期的重启

### DISCUSSION THREAD

**Discussion by 7y-9** [alibaba/ilogtail#1342] [Help] [answered]
**Describe the bug**
在user_yaml_config.d下新增和删除配置文件后，所有的yaml配置都会重启一遍。

**iLogtail Running Environment**
Please provide the following information:
centos 7.4

- ilogtail version:
  1.7.0
- Yaml configuration:

- ilogtail.LOG:
[ilogtail.LOG](https://github.com/alibaba/ilogtail/files/13390613/ilogtail.LOG)

- logtail_plugin.LOG:
[logtail_plugin1.LOG](https://github.com/alibaba/ilogtail/files/13390669/logtail_plugin1.LOG)

**Comment by yyuuttaaoo** [comment]
这个问题是已知的，当前因为设计原因会导致所有配置重启，在后面的版本迭代中会改进。
这个问题对你的使用场景造成了什么影响可以写一下

**Comment by 7y-9** [comment, marked as answer]
> 这个问题是已知的，当前因为设计原因会导致所有配置重启，在后面的版本迭代中会改进。 这个问题对你的使用场景造成了什么影响可以写一下

感谢回复。
场景是这样的，我们自定义了service_thrift input 。配置变更时插件重启，service_thrift重启后，老的input不会被停成功，新的input端口占用起不来；同时老input的正常在采集。
